source/adios2/toolkit/transport/file/FilePointer.cpp

```diff
@@ -48,16 +48,16 @@ void FilePointer::Open(const std::string &name, const OpenMode openMode)
 
     if (m_OpenMode == OpenMode::Write)
     {
-        m_File = std::fopen(name.c_str(), "w");
+        m_File = fopen(name.c_str(), "wb");
```

Comment:
So, it looks like the `#define` has changed across different versions of visual studio.  Some used `_CRT_SECURE_NO_WARNINGS` and others used `_CRT_SECURE_NO_DEPRECATE`.  There's also another set of deprecation warnings you can run into that are essentially the same thing that need the `_SCL` prefix to turn off.  Defining both should suffice:
```cpp
#ifdef _MSC_VER
#define _CRT_SECURE_NO_WARNINGS
#define _CRT_SECURE_NO_DEPRECATE
#define _SCL_SECURE_NO_WARNINGS
#define _SCL_SECURE_NO_DEPRECATE
#endif
```
It's a pretty common code block for cross-platform C and C++ to address this very problem.